py/dml/objects.py

```diff
@@ -216,8 +215,12 @@ def __init__(self, ident, site, parent, arraylens = (), idxvars = ()):
 
         self._confidential = False
 
-        self.uniq = len(dml.globals.objects) + 1
-        dml.globals.objects.append(self)
+        # during code generation, set to a unique positive integer,
+        # used as object ID, e.g. in the id_infos array.  This integer
+        # number is the index into object.Device.objects *plus one*;
+        # the plus one is because object ID 0 is reserved in runtime
+        # for an invalid reference.
+        self.uniq = None
```

Comment:
This belongs under `CompositeObject`, not `DMLObject`.

Reply:
This is `CompositeObject.__init__`.

Reply:
So it is, I failed at GitHub.